src/models/processor.cpp

```diff
@@ -198,8 +198,93 @@ std::unique_ptr<OrtValue> ProcessTensor<float, int64_t>(OrtxTensor* tensor, Ort:
   return tensor_value;
 }
 
+template <typename T>
+OrtxTensor* MakeOrtxTensor(Generators::Tensor* src) {
+  if (!src) {
+    throw std::runtime_error("Null tensor passed to MakeOrtxTensor");
+  }
+
+  T* data = const_cast<T*>(src->GetData<T>());
+  std::vector<int64_t> shape = src->GetShape();
+
+  auto* ort = new Ort::Custom::Tensor<T>(shape, data);
+  return reinterpret_cast<OrtxTensor*>(ort);
+}
+
+template <typename T>
+const OrtxTensor* MakeOrtxTensorConst(const Generators::Tensor* src) {
+  if (!src) {
+    throw std::runtime_error("Null tensor passed to MakeOrtxTensorConst");
+  }
+
+  const T* data = src->GetData<T>();
+  std::vector<int64_t> shape = src->GetShape();
+
+  auto* ort = new Ort::Custom::Tensor<T>(shape, const_cast<T*>(data));
+  return reinterpret_cast<const OrtxTensor*>(ort);
+}
+
+void* SplitSignalSegments(
+    const Generators::Tensor* input,
+    const Generators::Tensor* sr_tensor,
+    const Generators::Tensor* frame_ms_tensor,
+    const Generators::Tensor* hop_ms_tensor,
+    const Generators::Tensor* energy_threshold_db_tensor,
+    Generators::Tensor* output0) {
+  if (!input || !sr_tensor || !frame_ms_tensor || !hop_ms_tensor ||
+      !energy_threshold_db_tensor || !output0) {
+    throw std::runtime_error("Null tensor argument passed to OgaSplitSignalSegments");
+  }
+
+  const OrtxTensor* in_tensor = Generators::MakeOrtxTensorConst<float>(input);
```

Comment:
it seems there's a constructor for every cast - given that there aren't tons of calls for it (nor will there be the need), i feel there isn't reason to clutter the API with these constructors, and we can just use `reinterpret_cast<OrtxTensor*>` when needed (which is what the constructors do).